src/Algebra/Module/Properties.agda

```diff
@@ -0,0 +1,98 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Definitions of algebraic structures defined over some other
+-- structure, like modules and vector spaces
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Algebra.Module.Properties where
```

Comment:
Hmm hierarchies don't usually have `.Properties` file but instead have a `Properties.S` for every structure `S` (which are usually parameterised by an `S`).

For example `semiring⇒leftSemimodule` should live in `Algebra.Properties.Semiring` and `ring⇒rightModule` should live in `Algebra.Properties.Ring`

Reply:
Do you think `Algebra.Properties.LeftSemimodule` &c is the way to go? Also, would this imply simplifying the names of lemmas? E.g, `semiring⇒leftSemimodule` becomes `fromSemiring`, or something like this.

Reply:
This seems like putting a LOT in the module hierarchy, though: {left, right, neuter} * {semi, full}... that's a lot of modules.

Reply:
Yeah, this was also part of the reasoning. All of these lemmas are pretty similar, so it seems easier to keep them synchronised if they are together. Maybe we should think about how large this file is likely to get.

Reply:
> This seems like putting a LOT in the module hierarchy, though: {left, right, neuter} * {semi, full}... that's a lot of modules.

@wenkokke you're right, but it is the way the library is currently laid out (see `Algebra.Properties`). I think best to try and stay consistent. 

> Do you think Algebra.Properties.LeftSemimodule &c is the way to go? Also, would this imply simplifying the names of lemmas? E.g, semiring⇒leftSemimodule becomes fromSemiring, or something like this.

Yup sounds good to me.

Reply:
Hmm okay, new plan! Keep this module as is, but rename it to `Algebra.Module.Construct.Trivial`? As you suggested earlier @laMudri, we could then ditch the `semiring⇒` part of the `semiring⇒leftSemimodule` names?

Edit: ah, just seen that you suggested this as well. Great minds think alike :smile: